docs/parser.md

```diff
@@ -0,0 +1,502 @@
+| `flow`        | `ContentFlowComponent`        | `src/app/page/component/content-flow/`        |
+| `spacer`      | `ContentSpacerComponent`      | `src/app/page/component/content-spacer/`      |
+
+### Full Flow Example
+
+```
+XML:
+<page>
+  <content:paragraph>
+    <content:text text-color="#333333">Hello World</content:text>
+  </content:paragraph>
+</page>
+
+↓ (Kotlin Parser)
+
+TypeScript Object:
+Paragraph {
+  content: [
+    Text {
+      text: "Hello World",
+      textColor: Color(51, 51, 51)
+    }
+  ]
+}
+
+↓ (ContentRepeater)
+
+ContentParagraphComponent receives Paragraph
+  ↓
+ContentTextComponent receives Text
+  ↓
+Renders: <span style="color: #333333">Hello World</span>
+```
+
+## Event System
+
+The event system enables interactive navigation and actions within GodTools content.
+
+### EventId Structure
+
+Events are identified by an `EventId` object:
+
+```typescript
+class EventId {
+  namespace: string | null; // Optional namespace
+  name: string; // Event name
+}
+```
+
+Events can be specified in XML as:
+
+- `name` - Simple event (e.g., `next-page`)
+- `namespace:name` - Namespaced event (e.g., `state:selectedOption`)
+
+**Kotlin Definition**: `kotlin-mpp-godtools-tool-parser/module/parser-base/src/commonMain/kotlin/org/cru/godtools/shared/tool/parser/model/EventId.kt`
+
+### Special Event Namespaces
+
+#### `state:` Namespace
+
+Events with the `state:` namespace are **dynamic** and resolved at runtime from the parser State:
+
+```typescript
+// XML: events="state:selectedOption"
+// If State has: selectedOption = ["option-a", "option-b"]
+// Then resolve() returns: [EventId("option-a"), EventId("option-b")]
+
+eventId.resolve(state);
+```
+
+This allows content to trigger different navigation based on user choices stored in State.
+
+**Kotlin Implementation**: `kotlin-mpp-godtools-tool-parser/module/renderer-state/src/commonMain/kotlin/org/cru/godtools/shared/tool/parser/model/EventId+State.kt`
+
+#### `followup:` Namespace
+
+Special namespace for follow-up actions: `EventId.FOLLOWUP = EventId("followup", "send")`
+
+### Events on Content Elements
+
+Interactive content elements can trigger events when clicked:
+
+```typescript
+// Button with event
+<Button
+  text="Next"
+  events={[EventId("next-page")]}
+/>
+
+// Image with multiple events
+<Image
+  resource="image.png"
+  events={[EventId("zoom"), EventId("track-click")]}
+/>
+
+// Card with state-based navigation
+<Card
+  events={[EventId("state", "userChoice")]}
+/>
+```
+
+Components that support events:
+
+- **Button** (`src/app/page/component/content-button/content-button.component.ts`)
+- **Link** (`src/app/page/component/content-link/content-link.component.ts`)
+- **Image** (`src/app/page/component/content-image/content-image.component.ts`)
+- **Card** (`src/app/page/component/content-card/content-card.component.ts`)
+- **Animation** (for interactive animations)
+
+### Page Listeners
```

Comment:
the other places that use listeners off the top of my head:
- Animations
  - playListeners
  - stopListeners
- Tabs
  - each tab can have listeners to display that tab, this isn't really applicable to web rendering currently where tabs are flattened
- Tract Cards
  - support listeners and dismissListeners to jump to that card, not full applicable for web
- Tract Modals
  - same as Tract Cards